meta-facebook/yv35-gl/src/platform/plat_isr.c

```diff
@@ -13,16 +13,493 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#include <logging/log.h>
 
-#include "plat_isr.h"
+#include "hal_gpio.h"
 #include "hal_vw_gpio.h"
+#include "libipmi.h"
+#include "libutil.h"
+#include "snoop.h"
+#include "ipmb.h"
+#include "ipmi.h"
 #include "power_status.h"
-#include <logging/log.h>
+#include "kcs.h"
+
+#include "plat_i2c.h"
+#include "plat_isr.h"
+#include "plat_gpio.h"
+#include "plat_ipmb.h"
+#include "plat_sensor_table.h"
 
 LOG_MODULE_REGISTER(plat_isr);
 
+void send_gpio_interrupt(uint8_t gpio_num)
+{
+	ipmb_error status;
+	ipmi_msg msg;
+	uint8_t gpio_val;
+
+	gpio_val = gpio_get(gpio_num);
+	LOG_INF("Send gpio interrupt to BMC, gpio number(%d) status(%d)\n", gpio_num, gpio_val);
+
+	msg.data_len = 5;
+	msg.InF_source = SELF;
+	msg.InF_target = BMC_IPMB;
+	msg.netfn = NETFN_OEM_1S_REQ;
+	msg.cmd = CMD_OEM_1S_SEND_INTERRUPT_TO_BMC;
+
+	msg.data[0] = IANA_ID & 0xFF;
+	msg.data[1] = (IANA_ID >> 8) & 0xFF;
+	msg.data[2] = (IANA_ID >> 16) & 0xFF;
+	msg.data[3] = gpio_num;
+	msg.data[4] = gpio_val;
+
+	status = ipmb_read(&msg, IPMB_inf_index_map[msg.InF_target]);
+	if (status != IPMB_ERROR_SUCCESS) {
+		LOG_ERR("Failed to send GPIO interrupt event to BMC, gpio number(%d) status(%d)",
+			gpio_num, status);
+	}
+}
+
+static void SLP3_handler()
+{
+	common_addsel_msg_t sel_msg;
+	if (gpio_get(PWRGD_CPU_LVC3) == GPIO_LOW) {
+		sel_msg.InF_target = BMC_IPMB;
+		sel_msg.sensor_type = IPMI_OEM_SENSOR_TYPE_SYS_STA;
+		sel_msg.event_type = IPMI_EVENT_TYPE_SENSOR_SPECIFIC;
+		sel_msg.sensor_number = SENSOR_NUM_SYSTEM_STATUS;
+		sel_msg.event_data1 = IPMI_OEM_EVENT_OFFSET_SYS_VRWATCHDOG;
+		sel_msg.event_data2 = 0xFF;
+		sel_msg.event_data3 = 0xFF;
+		if (!common_add_sel_evt_record(&sel_msg)) {
+			LOG_ERR("VR watchdog timeout addsel fail");
+		}
+	}
+}
+
+K_WORK_DELAYABLE_DEFINE(SLP3_work, SLP3_handler);
+void ISR_SLP3()
+{
+	if (gpio_get(PWRGD_CPU_LVC3) == GPIO_LOW) {
+		LOG_INF("slp3\n");
+		k_work_schedule(&SLP3_work, K_MSEC(10000));
+		return;
+	}
+	if (k_work_cancel_delayable(&SLP3_work) != 0) {
+		LOG_ERR("[%s] Failed to cancel delayable work", __func__);
+	}
+}
+
+K_WORK_DELAYABLE_DEFINE(set_DC_on_5s_work, set_DC_on_delayed_status);
+K_WORK_DELAYABLE_DEFINE(set_DC_off_10s_work, set_DC_off_delayed_status);
+/*
+ * TODO :
+ * Implement "Read PMIC error when DC off" when I3C is ready.
+ */
+
+void Set_DC_status()
+{
+	set_DC_status(PWRGD_CPU_LVC3);
+
+	if (get_DC_status() == true) {
+		k_work_schedule(&set_DC_on_5s_work, K_SECONDS(DC_ON_5_SECOND));
+
+		if (k_work_cancel_delayable(&set_DC_off_10s_work) != 0) {
+			LOG_ERR("Cancel set dc off delay work fail");
+		}
+		set_DC_off_delayed_status();
+	} else {
+		set_DC_on_delayed_status();
+		k_work_schedule(&set_DC_off_10s_work, K_SECONDS(DC_OFF_10_SECOND));
+
+		/*
```

Comment:
Remove todo

Reply:
Removed.
Please review it again, thanks.